test/unit/math/prim/prob/poisson_binomial_test.cpp

```diff
@@ -0,0 +1,158 @@
+#include <stan/math/prim.hpp>
+#include <gtest/gtest.h>
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/math/distributions.hpp>
+#include <limits>
+#include <vector>
+
+using stan::math::poisson_binomial_lccdf;
+using stan::math::poisson_binomial_lcdf;
+using stan::math::poisson_binomial_lpmf;
+using vec = Eigen::Matrix<double, Eigen::Dynamic, 1>;
+
+static double inff = std::numeric_limits<double>::infinity();
+
+TEST(ProbDistributionsPoissonBinomial,
+     poisson_binomial_check_error_scalar_y_oob) {
+  vec theta(3);
+  theta << 0.5, 0.2, 0.7;
+
+  EXPECT_THROW(poisson_binomial_lpmf(-1, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lpmf(4, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lcdf(-1, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lcdf(4, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lccdf(-1, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lccdf(4, theta), std::domain_error);
+}
+
+TEST(ProbDistributionsPoissonBinomial, check_error_theta_is_not_prob) {
+  vec theta(3);
+  theta << 0.5, 0.2, 1.1;
+  vec theta2(3);
+  theta2 << 0.5, 0.2, inff;
+
+  EXPECT_THROW(poisson_binomial_lpmf(1, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lpmf(1, theta2), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lcdf(1, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lcdf(1, theta2), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lccdf(1, theta), std::domain_error);
+  EXPECT_THROW(poisson_binomial_lccdf(1, theta2), std::domain_error);
+}
+
+TEST(ProbDistributionsPoissonBinomial, check_error_vectorial_y_oob) {
+  vec theta(3);
+  theta << 0.5, 0.2, 0.1;
+  std::vector<int> ys1{-1, 2};
+  std::vector<int> ys2{4, 3};
+
+  EXPECT_THROW(poisson_binomial_lpmf(ys1, theta), std::domain_error);
```

Comment:
This is a bit odd that it throws here, but not on L24 of ```poisson_binomial_pmf_log_test```

Reply:
Not sure, if I misunderstand: it throws here because 4 is out of bounds (`size(theta) = 3`, so y can be max `3`. In L24  `std::vector<int> ys2{2, 2}` shouldn't throw.